cobrawap/__main__.py

```diff
@@ -361,38 +514,23 @@ def run_stage(profile=None, stage=None, extra_args=None, **kwargs):
     return None
 
 
-def run_block(block=None, block_args=None, block_help=False, **kwargs):
-    while not block:
-        block = input("Specify a block to execute (<stage_name>.<block_name>):")
-
-    stage, block  = re.split("\.|/|\s", block)[:2]
-    stages = get_setting('stages')
-
-    while not stage in stages.values():
-        print(f"Stage {stage} not found!\n"\
-              f"Available stages are: {', '.join(list(stages.values()))}.")
-        stage = input("Specify stage:")
-
-    block_dir = Path(get_setting('pipeline_path')) / stage / 'scripts'
-    available_blocks = [str(script.stem) for script in block_dir.iterdir()
-                        if not str(script.stem).startswith('_')]
+def run_block(stage=None, block=None, block_args=None, block_help=False, **kwargs):
+    # # allow for positional block argument
+    # if block is None and block_args and block_args[0][0] != "-":
+    #     block = block_args.pop(0)
 
-    while not block in available_blocks:
-        print(f"Block {block} is not found in {stage}!\n"\
-              f"Available blocks are: {', '.join(available_blocks)}.")
-        block = input("Specify block:")
+    stage = input_stage(stage=stage)
+    block = input_block(stage=None, block=block)
```

Comment:
Here is a bug preventing `stage` value to be properly passed to `input_block` function:
 
```suggestion
    block = input_block(stage=stage, block=block)
```